cloud/gke.go

```diff
@@ -0,0 +1,1038 @@
+package cloud
+
+import (
+	"os"
+	"io/ioutil"
+	"golang.org/x/oauth2/google"
+	gke "google.golang.org/api/container/v1"
+	"golang.org/x/net/context"
+	"strings"
+	"time"
+	"google.golang.org/api/googleapi"
+	"net/http"
+	"github.com/gin-gonic/gin"
+
+	banzaiUtils "github.com/banzaicloud/banzai-types/utils"
+	banzaiConstants "github.com/banzaicloud/banzai-types/constants"
+	banzaiTypes "github.com/banzaicloud/banzai-types/components"
+	banzaiSimpleTypes "github.com/banzaicloud/banzai-types/components/database"
+	"github.com/banzaicloud/banzai-types/database"
+	"encoding/base64"
+	"fmt"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/api/core/v1"
+	"k8s.io/api/rbac/v1beta1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"gopkg.in/yaml.v2"
+	"path/filepath"
+	"github.com/Azure/go-autorest/autorest/to"
+)
+
+var credentialPath string
+
+const (
+	statusRunning = "RUNNING"
+)
+
+const (
+	defaultNamespace = "default"
+	clusterAdmin     = "cluster-admin"
+	netesDefault     = "netes-default"
+)
+
+const googleAppCredential = "GOOGLE_APPLICATION_CREDENTIALS"
+
+func init() {
+	credentialPath = os.Getenv(googleAppCredential)
```

Comment:
Shouldn't we use viper.getString here? and put GOOGLE_APPLICATION_CREDENTIALS to config.toml.example?